hmiddleware/request_logger_test.go

```diff
@@ -0,0 +1,50 @@
+package hmiddleware
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func TestNewLogEntry(t *testing.T) {
+	req, _ := http.NewRequest(http.MethodGet, "/", nil)
+	req.Header.Set("X-Heroku-Robot", "I am a robot")
+	logger := logrus.New()
+	l := &StructuredLogger{
+		Logger: logger,
+	}
+
+	e := l.NewLogEntry(req)
+	if e == nil {
+		t.Error("error creating NewLogEntry.")
+	}
+}
+
+func TestLogEntryWrite(t *testing.T) {
+	req, _ := http.NewRequest(http.MethodGet, "/", nil)
+	logger := logrus.New()
+	l := &StructuredLogger{
+		Logger: logger,
+	}
+	e := l.NewLogEntry(req)
+	status := 200
+	bytes := 0
+	elapsed := time.Duration(100)
+
+	e.Write(status, bytes, elapsed)
+}
+
+func TestLogEntryPanic(t *testing.T) {
+	req, _ := http.NewRequest(http.MethodGet, "/", nil)
+	logger := logrus.New()
```

Comment:
This is a good case for using the [testlog package](https://godoc.org/github.com/heroku/x/testing/testlog#NewNullLogger). It looks like there aren't any examples there, but your test might look something like:

```go
logger, hook := testlog.NewNullLogger()
l := &StructuredLogger{
	Logger: logger,
}
e := l.NewLogEntry(req)

err := errors.New("test error")
stack := "foo.go:123"
e.Panic(err, []byte(stack))

hook.CheckContained(t, `error="test error"`, `stack="foo.go:123"`)
```

Reply:
ah yeah, I was using that for the old logger middleware that was removed in this PR.

@jboursiquot-sfdc's opinion was that testing what the logs themselves contain was looking too much at the implementation details of the methods we are adding (rather than the output of the methods) but it does feel odd to not have any tests looking at the log output. 🤔

Reply:
@jessieay Testing the log output here does make sense since we're checking that `e.Panic` should emit certain key/value pairs. The one where it wasn't warranted was for the `NewLogEntry` test since we were only checking that a `LogEntry` was indeed returned.

Reply:
note to self: look into `hook.LastEntry` so you can look and see if the fields match what you expect

Reply:
added log output info in specs at a67c99e